storage/connectors/redis/src/main/java/feast/storage/connectors/redis/common/RedisHashDecoder.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2020 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.storage.connectors.redis.common;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.Timestamp;
+import feast.proto.serving.ServingAPIProto;
+import feast.proto.types.ValueProto;
+import feast.storage.api.retriever.Feature;
+import io.lettuce.core.KeyValue;
+import java.util.*;
+
+public class RedisHashDecoder {
+
+  /**
+   * Converts all retrieved Redis Hash values based on EntityRows into {@link Feature}
+   *
+   * @param redisHashValues retrieved Redis Hash values based on EntityRows
+   * @param isTimestampMap map to determine if Redis Hash key is a timestamp field
+   * @param byteToFeatureReferenceMap map to decode bytes back to FeatureReference
+   * @return List of {@link Feature}
+   * @throws InvalidProtocolBufferException
+   */
+  public static List<Optional<Feature>> retrieveFeature(
```

Comment:
`isTimestampMap` seems redundant here, since you're passing `timestampPrefix` anyway.
so whether element is timestamp can be just checked by
`Arrays.toString(redisValueK).startsWith(timestampPrefix)`